Source/Core/Core/HW/WiimoteReal/WiimoteReal.cpp

```diff
@@ -916,7 +916,9 @@ static void TryToConnectBalanceBoard(std::unique_ptr<Wiimote> wm)
 static void HandleWiimoteDisconnect(int index)
 {
   Core::RunAsCPUThread([index] {
-    g_wiimotes[index] = nullptr;
+    // The Wii Remote object must exist through the call to UpdateSource
+    // to prevent WiimoteDevice from having a dangling HIDWiimote pointer.
+    const auto temp_real_wiimote = std::move(g_wiimotes[index]);
```

Comment:
I guess alternatively we could have just swapped these calls?

Reply:
No. Unfortunately `UpdateSource` gives `WiimoteDevice` a new pointer by grabbing it from `g_wiimotes`.

Reply:
Isn't `UpdateSource` going to reach here: https://github.com/dolphin-emu/dolphin/blob/0029ca84b07f44e3ce8c80a44fa0285a966b34d3/Source/Core/Core/HW/Wiimote.cpp#L75 and `.get()` the moved-from index?

Reply:
ok, the comment is a bit confusing to me for some reason. anyway lgtm